mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping

```diff
@@ -1,12 +1,15 @@
 CLASS net/minecraft/class_1688 net/minecraft/entity/vehicle/AbstractMinecartEntity
+	FIELD field_24464 DISMOUNT_FREE_Y_SPACES_NEEDED Lcom/google/common/collect/ImmutableMap;
 	FIELD field_7655 clientYVelocity D
 	FIELD field_7656 clientZVelocity D
 	FIELD field_7657 clientPitch D
 	FIELD field_7658 clientXVelocity D
 	FIELD field_7659 clientYaw D
+	FIELD field_7660 isYawFlipped Z
 	FIELD field_7661 CUSTOM_BLOCK_OFFSET Lnet/minecraft/class_2940;
 	FIELD field_7662 clientZ D
 	FIELD field_7663 DAMAGE_WOBBLE_TICKS Lnet/minecraft/class_2940;
+	FIELD field_7664 RAIL_SHAPE_TO_DIRECTION_OFFSETS Ljava/util/Map;
```

Comment:
:x:
not really direction offsets, but like two ends for lerping.

Reply:
Any other suggested name? `RAIL_SHAPE_TO_END_COORDINATES`? `RAIL_SHAPE_TO_DIRECTION_VECTORS`?

Reply:
`LERP_ENDS` or `LERP_OFFSETS` imo, interpolation ends (xyz offsets) are like unique to each rail direction

Reply:
Also what do you think of my analysis

Reply:
I don't think lerping is a good name. That's an algorithm which uses these, not what they mean.

Reply:
That is exactly lerping logic, i.e. interpolation logic.

Reply:
Can you describe minecart lerping?

Reply:
It is just used for interpolating a minecart's position on a certain type of rail shape, simple as that

Reply:
Then maybe "minecart position lerping" would be more specific

Reply:
your "minecart" and "position" are both useless words that can be inferred from context. maybe `SNAPPING_INTERPOLATIONS` if you really like your `snapToRail` name

Reply:
As I see it the main purpose is to define which direction the minecart moves on each rail shape. `SNAPPING_INTERPOLATIONS` does not convey that information - nor does the word "lerping". `snapToRail` can be renamed to `getClosestPositionOnRail` if you don't like verb `snap`.

Reply:
it is the direction of the rail shape as defined by two points in 3d-space.